src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5PoNCreator.java

```diff
@@ -0,0 +1,712 @@
+     *     targets, and the right one is the corresponding target factors.
+     * </p>
+     *
+     * @param readCounts the input read-counts.
+     * @param targetFactorPercentileThreshold the minimum median count percentile under which targets are not considered useful.
+     * @return never {@code null}.
+     */
+    @VisibleForTesting
+    static Pair<ReadCountCollection, double[]> subsetReadCountsToUsableTargets(final ReadCountCollection readCounts,
+                                                                               final double targetFactorPercentileThreshold, final Logger logger) {
+        final double[] targetFactors = calculateTargetFactors(readCounts);
+        final Percentile percentile = new Percentile(targetFactorPercentileThreshold);
+        final double threshold = percentile.evaluate(targetFactors);
+        final List<Target> targetByIndex = readCounts.targets();
+        final Set<Target> result = IntStream.range(0, targetFactors.length).filter(i -> targetFactors[i] >= threshold)
+                .mapToObj(targetByIndex::get)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+        if (result.size() == targetByIndex.size()) {
+            logger.info(String.format("All %d targets are kept", targetByIndex.size()));
+            return new ImmutablePair<>(readCounts, targetFactors);
+        } else {
+            final int discardedCount = targetFactors.length - result.size();
+            logger.info(String.format("Discarded %d target(s) out of %d with factors below %.2g (%.2f percentile)", discardedCount, targetFactors.length, threshold, targetFactorPercentileThreshold  ));
+            final double[] targetFactorSubset = DoubleStream.of(targetFactors).filter(i -> i >= threshold).toArray();
+            return new ImmutablePair<>(readCounts.subsetTargets(result), targetFactorSubset);
+        }
+    }
+
+    /**
+     * Reads the read-counts from a file.
+     * <p>
+     *     The read-counts are subset
+     * </p>
+     *
+     * @param inputFile the input file containing the read-counts.
+     * @param targets the target collection to subset to.
+     * @return never {@code null}.
+     */
+    private static ReadCountCollection readReadCountsFromFile(final File inputFile, final TargetCollection<Target> targets) {
+        try {
+            return ReadCountCollectionUtils.parse(inputFile, targets, true);
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputFile);
+        }
+    }
+
+    /**
+     * Calculate the target factors:
+     *
+     * <p>These are the median coverage per target</p>.
+     * @param readCounts the input read counts.
+     * @return never {@code null}, with as many elements as targets in {@code readCounts}.
+     */
+    private static double[] calculateTargetFactors(final ReadCountCollection readCounts) {
```

Comment:
Since it comes up so much, you should extract methods for getting arrays of row medians and column medians from a matrix.